### ISSUE TITLE
CodePipeline Custom actions cause failure

### ISSUE THREAD

**Issue by palsarma** [awslabs/cfn-python-lint#397]
$ cfn-lint --version
cfn-lint 0.7.4 

$ python --version
Python 3.6.5

*Description of issue.*
cfn-lint fails rule E2541 if Custom actions are specified for CodePipeline resource.

> E0002 Unknown exception while processing rule E2541: unhashable type: 'dict_node'

The exception is happening here:
https://github.com/awslabs/cfn-python-lint/blob/a1c1f4688f4ca0a8c12d85e239752193499d06f3/src/cfnlint/rules/resources/codepipeline/CodepipelineStageActions.py#L114

Please provide as much information as possible:
* Sample Template :
```
Resources:
  Pipeline:
    Type: AWS::CodePipeline::Pipeline
    Properties:
      ArtifactStore:
        Type: S3
        Location: !ImportValue PipelineBucket
      Name: BBSourceActionDemo
      RoleArn: !ImportValue PipelineRole.Arn
      Stages:
        - Name: 'Source'
          Actions:
            - Name: 'Source'
              ActionTypeId:
                Category: 'Source'
                Owner: 'Custom'
                Version: !ImportValue SourceActionVersion
                Provider: !ImportValue SourceActionProvider
              OutputArtifacts:
                - Name: MyAppSources
              Configuration:
                Branch: !ImportValue Branch
                BitBucketUrl: !ImportValue BitBucketUrl
              RunOrder: 1
        - Name: 'Build'
          Actions:
            - Name: 'CodeBuild'
              ActionTypeId:
                Category: 'Build'
                Owner: 'AWS'
                Version: '1'
                Provider: 'CodeBuild'
              InputArtifacts:
                - Name: MyAppSources
              OutputArtifacts:
                - Name: MyAppBuilt
              Configuration:
                ProjectName: !ImportValue CodeBuild
              RunOrder: 1
```

**Comment by kddejong** [comment]
Working on a fix for this.  Question for you.

According to the documentation. https://docs.aws.amazon.com/codepipeline/latest/userguide/reference-pipeline-structure.html

`For all currently supported action types, the only valid version string is "1".` I thought the version may represent something else with custom actions.  Are you using 1 for the Import value? does it work if it isn't 1?

**Comment by kddejong** [comment]
This is should be resolved with v0.8.1
